C-Sharp/VectorDistanceCalculation/PE11VectorDistanceCalculation/Point2D.cs

```diff
@@ -6,35 +6,49 @@
 
 namespace PE11VectorDistanceCalculation
 {
+    //Comments in this class apply for both objects
+    //Why have two objects? You can accomplish everything with one object and an extra property z here
+    //if you remove the methods from this class it can become a struct
     class Point2D : IEquatable<Point2D>
     {
+        //I like this class being immutable but no reason for the private set
         public int X { get; private set; }
         public int Y { get; private set; }
-
         public Point2D()
         {
+            //two problems with assigning X, Y here. 
+            //1. theyre value types, what is the default for the value types?
+            //2. what does this constructor accomplish?
             X = 0;
             Y = 0;
         }
 
+        //How do I set this object?
+        //Both constructors have internal assignments and no way for you to set the vector
+        //but I may not understand the assignment
         public Point2D(Random random, int min = 1, int max = 101)
         {
             X = random.Next(min, max);
             Y = random.Next(min, max);
         }
 
+        //This method doesnt belong here. This method belongs in a static utility class 
+        //Helper classes can maintain state
+        //Utility classes have static methods and only operate on parameters and return something
         public double CalculateDistanceTo(Point2D point)
         {
             return Math.Sqrt(Math.Pow((point.X - X), 2) + Math.Pow((point.Y - Y), 2));
         }
 
+        //good
         public override string ToString()
         {
             return $"X: {X}, Y: {Y}";
         }
 
         public bool Equals(Point2D other)
         {
+            //idk but I feel like there is a bug here
```

Comment:
It turns out this could become a very expensive operation...  If we just have one class, I don't even know how I would even implement it.  The purpose was to be able to compare one Point to another to see if they are the same...

Reply:
You said that in the email too. I am very curious to see why you think that? Have you tried doing any performance monitoring for this?

Reply:
Not a true performance test, but a VERY significant difference when I generated the random numbers within Point class (instead of the collection classes).  Once I removed the Equals method, I was able to get instant results...  I could add the modified one to the repo for you to have a look.

Reply:
ok do that.

Reply:
Just uploaded it!